test/Microsoft.TestPlatform.AdapterUtilities.UnitTests/ManagedNameUtilities/FindMethodExtensions.cs

```diff
@@ -54,7 +52,7 @@ internal static MethodInfo FindMethod(this Type type, string signature)
         }
 
         Debug.Assert(candidates.Any() && !candidates.Skip(1).Any());
-        return candidates.SingleOrDefault();
+        return candidates.Single();
```

Comment:
Looks like we were using a lot XXXOrDefault methods even when we know we have one element (see debug assert just before) or also doing lots of as cast when we know this won't be null.